bee-message/src/payload/milestone/mod.rs

```diff
@@ -46,22 +39,18 @@ impl From<CryptoError> for MilestoneValidationError {
     }
 }
 
-#[derive(Clone, Debug, Eq, PartialEq, Packable)]
-#[cfg_attr(feature = "serde1", derive(serde::Serialize, serde::Deserialize))]
-#[repr(transparent)]
-struct Signature(
-    #[cfg_attr(feature = "serde1", serde(with = "serde_big_array::BigArray"))] [u8; MilestonePayload::SIGNATURE_LENGTH],
-);
-
 pub(crate) type SignatureCount =
     BoundedU8<{ *MilestonePayload::SIGNATURE_COUNT_RANGE.start() }, { *MilestonePayload::SIGNATURE_COUNT_RANGE.end() }>;
 
 /// A payload which defines the inclusion set of other messages in the Tangle.
-#[derive(Clone, Debug, Eq, PartialEq)]
+#[derive(Clone, Debug, Eq, PartialEq, Packable)]
 #[cfg_attr(feature = "serde1", derive(serde::Serialize, serde::Deserialize))]
+#[packable(unpack_error = Error)]
 pub struct MilestonePayload {
     essence: MilestoneEssence,
-    signatures: VecPrefix<Signature, SignatureCount>,
+    #[packable(verify_with = verify_signatures)]
+    #[packable(unpack_error_with = |e| Error::MilestoneInvalidSignatureCount(e.into_prefix_err().into()))]
+    signatures: VecPrefix<Ed25519Signature, SignatureCount>,
```

Comment:
Just making sure: Does that setup still account for the 1 byte _Signature Type_ from the TIP?

Reply:
Good catch! No it doesn't